### PR TITLE
[rfr] Add support for Sensu enterprise

### PR DESCRIPTION
**Pull request by haphan**
Todo:  Add support for `sensu-enterprise`for following distributions
- ✅  CentOS/RHEL  
- ➖  Ubuntu

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -66,6 +82,8 @@ sensu_ssl_server_cacert: "{{ sensu_ssl_tool_base_path }}/sensu_ca/cacert.pem"
 sensu_ssl_server_cert: "{{ sensu_ssl_tool_base_path }}/server/cert.pem"
 sensu_ssl_server_key: "{{ sensu_ssl_tool_base_path }}/server/key.pem"
 sensu_ssl_tool_version: "0.23"
+dynamic_data_store: "{{ playbook_dir }}/data/store"
```
Comment (smuth4) [inline defaults/main.yml:87]: Didn't know `playbook_dir` was a thing, 👍

File: tasks/dashboard.yml [hunk 2]
```diff
@@ -2,5 +2,8 @@
 # tasks/dashboard.yml: Deployment of the Uchiwa dashboard
   - include: "{{ ansible_distribution }}/dashboard.yml"
 
-  - name: Ensure Uchiwa server service is running
-    service: name=uchiwa state=started enabled=yes
+  - name: Ensure Uchiwa/Sensu Enterprise Dashboard server service is running
+    service:
+      name: {{ uchiwa_service_name if not se_enterprise else sensu_enterprise_dashboard_service_name }}
```
Comment (smuth4) [inline tasks/dashboard.yml]: This value needs to be quoted.

File: tasks/common.yml [hunk 3]
```diff
@@ -21,7 +21,7 @@
     notify:
       - restart sensu-server service
       - restart sensu-api service
-      - restart sensu-client service
```
Comment (smuth4) [inline tasks/common.yml]: `sensu-client` will still need to be restarted for non-enterprise users on all of these (and for enterprise users as well I think?).

File: tasks/server.yml [hunk 4]
```diff
@@ -16,7 +16,8 @@
     static: false
 
   - name: Ensure Sensu server service is running
-    service: name=sensu-server state=started enabled=yes
+    service: name={{ sensu_server_service_name if not se_enterprise else sensu_enterprise_service_name }} state=started enabled=yes
 
   - name: Ensure Sensu API service is running
     service: name=sensu-api state=started enabled=yes
+    when: not se_enterprise
```
Comment (smuth4) [inline tasks/server.yml]: Gonna be a stickler and ask for a newline at EOF here 😀

File: defaults/main.yml [hunk 5]
```diff
@@ -39,7 +55,7 @@ sensu_api_password: secret
 sensu_api_uchiwa_path: ''
 sensu_api_timeout: 5000
 sensu_client_config: client.json.j2
-sensu_rabbitmq_config: rabbitmq.json.j2
+sensu_rabbitmq_config: sensu-rabbitmq.json.j2
```
Comment (smuth4) [inline defaults/main.yml:60]: What is this trying to be consistent with? None of the other config files use a `sensu-` prefix.

File: templates/rabbitmq.config.j2 [hunk 6]
```diff
@@ -1,10 +1,14 @@
 [
     {rabbit, [
+    {% if sensu_ssl_manage_certs %}
```
Comment (smuth4) [inline templates/rabbitmq.config.j2]: `sensu_ssl_manage_certs` was requested specifically by someone that wanted SSL to be used, but didn't want the cert files to be handled by this role. Please use a different variable or get rid of this change altogether.

File: templates/sensu-rabbitmq.json.j2 [hunk 7]
```diff
@@ -1,9 +1,11 @@
 {
   "rabbitmq": {
+    {% if sensu_ssl_manage_certs %}
```
Comment (smuth4) [inline templates/sensu-rabbitmq.json.j2]: Same here regarding `sensu_ssl_manage_certs`.

**Review by smuth4 — CHANGES_REQUESTED** [review]

File: defaults/main.yml (hunk 5 above)
Comment (haphan) [reply defaults/main.yml:60]: Sensu related template files should have prefix `sensu-` as seen in https://github.com/sensu/sensu-ansible/blob/master/tasks/common.yml#L19.

File: tasks/common.yml (hunk 3 above)
Comment (haphan) [reply tasks/common.yml]: `sensu-client` service may not exist in sensu master node for enterprise users (`sensu-enterprise` only comprises of `sensu-server` and `sensu-api` [Reference)](https://sensuapp.org/docs/0.28/platforms/sensu-on-rhel-centos.html#sensu-enterprise). 

Restart a non-existing service may fail the playbook. Besides, I don't know if it is a common installation that sensu master also monitors itself as a client.

Comment (haphan) [reply tasks/common.yml]: What do you think of a new variable `sensu_client` to deploy sensu client? @smuth4

**Comment by smuth4** [comment]
👍 to the update, the only things remaining are the `sensu_ssl_manage_certs` thing and a rebase (should only be documentation changes I think).

**Comment by haphan** [comment]
@smuth4  Can you review again? Thanks

**Review by smuth4 — APPROVED** [review]

**Comment by smuth4** [comment]
Thanks for all the work!

**Pull request merged**